src/overlays/actors/ovl_Boss_Goma/z_boss_goma.c

```diff
@@ -21,115 +111,1836 @@ const ActorInit Boss_Goma_InitVars = {
+    dList = Graph_Alloc(gfxCtx, 1 * sizeof(Gfx));
+    dListHead = dList;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_80919A40.s")
+    gSPEndDisplayList(dList++);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_80919C38.s")
+    return dListHead;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_80919E80.s")
+Gfx* BossGoma_NoBackfaceCullingDlist(GraphicsContext* gfxCtx) {
+    Gfx* dList;
+    Gfx* dListHead;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_80919F8C.s")
+    dList = Graph_Alloc(gfxCtx, 4 * sizeof(Gfx));
+    dListHead = dList;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091A16C.s")
+    gDPPipeSync(dList++);
+    // CVG_X_ALPHA | ALPHA_CVG_SEL may or may not help achieve the decaying textures effect
+    gDPSetRenderMode(dList++, G_RM_PASS, G_RM_AA_ZB_TEX_EDGE2);
+    gSPClearGeometryMode(dList++, G_CULL_BACK);
+    gSPEndDisplayList(dList++);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091A2E8.s")
+    return dListHead;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/BossGoma_Update.s")
+void BossGoma_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BossGoma* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091A560.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_goma.c", 4991);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091A9E8.s")
+    func_80093D18(globalCtx->state.gfxCtx);
+    Matrix_Translate(0.0f, -4000.0f, 0.0f, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091ABEC.s")
+    if (this->noBackfaceCulling) {
+        gSPSegment(oGfxCtx->polyOpa.p++, 0x08, BossGoma_NoBackfaceCullingDlist(globalCtx->state.gfxCtx));
+    } else {
+        gSPSegment(oGfxCtx->polyOpa.p++, 0x08, BossGoma_EmptyDlist(globalCtx->state.gfxCtx));
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091AC18.s")
+    SkelAnime_Draw(globalCtx, this->skelanime.skeleton, this->skelanime.limbDrawTbl, BossGoma_OverrideLimbDraw,
+                   BossGoma_PostLimbDraw, &this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/BossGoma_Draw.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_goma.c", 5012);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091ADA8.s")
+void BossGoma_SpawnChildGohma(BossGoma* this, GlobalContext* globalCtx, s16 i) {
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_GOMA, this->lastTailLimbWorldPos.x,
+                       this->lastTailLimbWorldPos.y - 50.0f, this->lastTailLimbWorldPos.z, 0, i * 0x5555, 0, i);
+    // @bug (maybe) if child gohma fails to spawn this softlocks due to the lack of a != NULL check?
+    this->childrenGohmaState[i] = 1;
```

Comment:
would this softlock though? dont think so but might be missing something

Reply:
Gohma stays on the ceiling as long as all gohmas aren't defeated (`this->childrenGohmaState[i] == -1` iirc for all three gohmas), and the child gohma actors are responsible for setting the state back to -1 (or whatever value)

Reply:
oh gotcha, neat